addons/web/static/src/views/fields/badge/badge_field.js

```diff
@@ -42,8 +46,11 @@ export const badgeField = {
     component: BadgeField,
     displayName: _t("Badge"),
     supportedTypes: ["selection", "many2one", "char"],
-    extractProps: ({ decorations }) => {
-        return { decorations };
+    extractProps: ({ decorations }, dynamicInfo) => {
+        return {
+            decorations,
+            colorIndex: dynamicInfo.context.color_index,
```

Comment:
Having this set in the context doesn't really make sense. It's simply an option for the widget, so it should be in the `options` attribute. I would suggest an option `color_field` (like we already have for many2many_tags), which is the name of a field. Then in the component, you can access `this.props.record.data[this.props.colorField]` to get the color index (assuming the colorField props is given obviously).

Reply:
Also, whatever change you make in the framework, it should be tested

Reply:
The option should also be documented in `supportedOptions` (grep in the code for examples). This is used by studio to make the option available there.